File: zig/fib.zig
const std = @import("std");

fn fib(n: i32) i32 {
    if (n <= 1) return n;
    return fib(n - 1) + fib(n - 2);
}

pub fn main() !void {
    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
    defer _ = gpa.deinit();
    const allocator = gpa.allocator();
    
    const args = try std.process.argsAlloc(allocator);
    defer std.process.argsFree(allocator, args);
    
    if (args.len < 2) {
        std.debug.print("Usage: fib <n>\n", .{});
        return;
    }
    
    const n = try std.fmt.parseInt(i32, args[1], 10);
    var timer = std.time.Timer.start() catch unreachable;
    const result = fib(n);
    const elapsed = timer.read();
    
    std.debug.print("Zig: fib({}) = {}\n", .{ n, result });
    std.debug.print("Time: {}ms\n", .{elapsed / 1_000_000});
}

File: java/Fib.java
public class Fib {
    public static int fib(int n) {
        if (n <= 1) return n;
        return fib(n - 1) + fib(n - 2);
    }
    
    public static void main(String[] args) {
        if (args.length == 0) {
            System.out.println("Usage: java Fib <n>");
            return;
        }
        
        int n = Integer.parseInt(args[0]);
        long startTime = System.currentTimeMillis();
        int result = fib(n);
        long endTime = System.currentTimeMillis();
        
        System.out.println("Java: fib(" + n + ") = " + result);
        System.out.println("Time: " + (endTime - startTime) + "ms");
    }
}